ecosystem_tests/dorkl.py

```diff
@@ -765,3 +794,385 @@ def basic_blueprint_test(blueprint_file_name,
+
+
+def second_invocation_test_path(blueprint_file_name,
+                                test_name,
+                                on_second_invoke,
+                                inputs=None,
+                                timeout=None,
+                                uninstall_on_success=True
+                                ):
+    """
+    Handle blueprint test path in second test invocation depends on
+    on_second_invoke value.
+    :param test_name:
+    :param on_second_invoke: Should be one of: resume,rerun,update.
+    :param inputs:
+    :param timeout:
+
+    """
+    logger.debug('on second_invocation_test_path')
+    if on_second_invoke == 'resume':
+        # TODO: Ask Tramell if we want to resume only install workflow
+        exec_id = find_install_execution_to_resume(test_name)
+        logger.debug('execution to resume: {id}'.format(id=exec_id))
+        try:
+            logger.info('resuming...')
+            executions_resume(exec_id, timeout)
+        except EcosystemTimeout:
+            # Give 5 seconds grace.
+            executions_list(test_name)
+            wait_for_execution(test_name, 'install', 10)
+        else:
+            logger.debug('inside the else in resume!!')
+            wait_for_execution(test_name, 'install', timeout)
+    elif on_second_invoke == 'rerun':
+        logger.info('Installing...')
+        try:
+            executions_list(test_name)
+            executions_start('install', test_name, timeout)
+        except EcosystemTimeout:
+            # Give 5 seconds grace.
+            executions_list(test_name)
+            wait_for_execution(test_name, 'install', 10)
+        else:
+            wait_for_execution(test_name, 'install', timeout)
+    elif on_second_invoke == 'update':
+        logger.info('updating deployment...')
+        try:
+            logger.info('Blueprints list: {0}'.format(
+                cloudify_exec('cfy blueprints list')))
+            update_bp_name = test_name + '_update'
+            blueprints_upload(blueprint_file_name, update_bp_name)
+            deployment_update(test_name,
+                              update_bp_name,
+                              inputs,
+                              timeout)
+        except EcosystemTimeout:
+            # Give 5 seconds grace.
+            executions_list(test_name)
+            wait_for_execution(test_name, 'update', 10)
+        else:
+            wait_for_execution(test_name, 'update', timeout)
+    if uninstall_on_success:
+        logger.info('Uninstalling...')
+        executions_start('uninstall', test_name, timeout)
+        wait_for_execution(test_name, 'uninstall', timeout)
+        try:
+            deployment_delete(test_name)
+            blueprints_delete(test_name)
+            if on_second_invoke == 'update':
+                blueprints_delete(update_bp_name)
+        except Exception as e:
+            logger.info('Failed to delete blueprint, '
+                        '{0}'.format(str(e)))
```

Comment:
For each group of commands under `if` and `elif` put the code in a separate function, for example, lines 970 - 979 should all be in one function that is called after `elif`. If you are worried about making `dorkl.py` too big, maybe we need to create a new module and organize everything better like: 
```
- ecosystem_tests
    - dorkl
       - __init__.py # Import popular methods like `basic_blueprint_test` and `prepare_secrets` here.
        - cloudify_api.py # for like `blueprints_upload`, `deployments_create`, `executions_list`, etc.
        - commands.py # for `handle_process`, `docker_exec`, `cloudify_exec`, etc.
        - testing.py # for stuff like `basic_blueprint_test` `basic_blueprint_test_dev` `prepare_commands`.
```